app/src/main/java/com/gophillygo/app/activities/MapsActivity.java

```diff
@@ -88,38 +105,120 @@ private void panToCurrentLocation() {
+            AttractionInfo attractionInfo = (AttractionInfo)  marker.getTag();
+            showPopup(attractionInfo);
+            Log.d(LOG_LABEL, "Clicked marker for attraction: " + attractionInfo.getAttraction().getId());
+            return false;
+        });
+    }
+
+    public void openDetail(Attraction attraction) {
+        Intent intent;
+        String idKey;
+        if (attraction.isEvent()) {
+            intent = new Intent(this, EventDetailActivity.class);
+            idKey = EventDetailActivity.EVENT_ID_KEY;
+        } else {
+            intent = new Intent(this, PlaceDetailActivity.class);
+            idKey = PlaceDetailActivity.DESTINATION_ID_KEY;
         }
-        return true;
+        long attractionId = attraction.getId();
+        intent.putExtra(idKey, attractionId);
+        startActivity(intent);
+    }
+
+    private void loadMarkers() {
+        if (markers == null) {
+            markers = new ArrayList<>(attractions.size());
+            for (T attractionInfo : attractions) {
+                if (attractionInfo.getLocation() != null) {
+                    DestinationLocation location = attractionInfo.getLocation();
+                    LatLng latLng = new LatLng(location.getY(), location.getX());
+                    float opacity = filterMatches(attractionInfo) ? 1f : 0.5f;
+                    Marker marker = googleMap.addMarker(new MarkerOptions()
+                            .position(latLng)
+                            .icon(markerIcon)
+                            .alpha(opacity)
+                    );
+                    marker.setTag(attractionInfo);
+                    markers.add(marker);
+                }
+            }
+        } else {
+            for (Marker marker : markers) {
+                float opacity = filterMatches((T) marker.getTag()) ? 1f : 0.5f;
+                marker.setAlpha(opacity);
+            }
+        }
+    }
+
+    private void showPopup(AttractionInfo attractionInfo) {
+        popupBinding.setAttractionInfo(attractionInfo);
+        popupBinding.setAttraction(attractionInfo.getAttraction());
+        popupBinding.notifyPropertyChanged(BR.attraction);
+        popupBinding.notifyPropertyChanged(BR.attractionInfo);
+
+        // Need to set map padding so "Google" logo is above popup, but we need to wait until the
+        // popup is visible in order to measure it's height.
+        final Handler handler = new Handler();
+        handler.postDelayed(() -> {
+            googleMap.setPadding(0, 0, 0, 25 + popupBinding.mapPopupCard.getHeight());
+        }, 30);
+    }
+
+    private BitmapDescriptor vectorToBitmap(@DrawableRes int id) {
+        Drawable vectorDrawable = ResourcesCompat.getDrawable(getResources(), id, null);
+        assert vectorDrawable != null;
+        Bitmap bitmap = Bitmap.createBitmap(vectorDrawable.getIntrinsicWidth(),
+                vectorDrawable.getIntrinsicHeight(), Bitmap.Config.ARGB_8888);
```

Comment:
This needs commenting. What is that config value constant for?

Reply:
You need to pass a config to `createBitmap(...)`, but per the docs they recommend using `ARGB_8888` whenever possible.